newsapi/app/src/main/java/com/example/newsapi/MainActivity.kt

```diff
@@ -1,69 +0,0 @@
-package com.example.newsapi
-
-import androidx.appcompat.app.AppCompatActivity
-import android.os.Bundle
-import androidx.recyclerview.widget.LinearLayoutManager
-import com.android.volley.Request
-import com.android.volley.Response
-import com.android.volley.toolbox.JsonObjectRequest
-import com.example.fuckin.News
-import com.example.fuckin.NewsItemClicked
-import com.example.fuckin.newslistadapter
-import com.example.fuckin.singleton
-import kotlinx.android.synthetic.main.activity_main.*
-import org.json.JSONObject
-
-
-class MainActivity : AppCompatActivity(), NewsItemClicked {
-
-
-    private lateinit var madapter: newslistadapter
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-        rvnews.layoutManager= LinearLayoutManager(this)
-        fetchData()
-        madapter =newslistadapter(this)
-        rvnews.adapter=madapter
-    }
-    private fun fetchData()
-    {
-        val url= "https://newsapi.org/v2/everything?q=tesla&from=2022-12-06&sortBy=publishedAt&apiKey=0552a96fa9db4a31a1b2c99ab89c0b7c"
-
-        val jsonObjectRequest = JsonObjectRequest(Request.Method.GET, url, null,
-            Response.Listener {
-
-                val newsJsonArray=it.getJSONArray("articles")//taking the json array from our api;
-                val newsArray=ArrayList<News>()
-                for( i in 0 until 50)
-                {
-                    val newJsonObject=newsJsonArray.getJSONObject(i)
-                    val news=News(newJsonObject.getString("title"),newJsonObject.getString("author"),
-                        newJsonObject.getString("url")
-                        ,newJsonObject.getString("imageUrl"))
-                    newsArray.add(news)
-
-
-
-                }
-                madapter.udatedNews(newsArray)
-
-            },
-            Response.ErrorListener {
```

Comment:
handle error show this image in case of error - https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.businesstoday.in%2Flatest%2Ftrends%2Fstory%2Fzomato-down-users-complain-of-not-being-able-to-order-food-325715-2022-03-11&psig=AOvVaw3qYuZ1J4ZVLG2T43UiuuT6&ust=1673705302484000&source=images&cd=vfe&ved=0CBAQjRxqFwoTCJCP56HcxPwCFQAAAAAdAAAAABAE